sgpt/make_prompt.py

```diff
@@ -1,86 +1,62 @@
-IMPORTANT: Provide full solution. Make sure syntax is correct.
-Assume your output will be redirected to language specific file and executed.
-For example Python code output will be redirected to code.py and then executed python code.py.
-
-Follow all of the above rules.
-This is important you MUST follow the above rules.
-There are no exceptions to these rules.
-You must always follow them. No exceptions.
-
-Request: """
-
-
-def shell(question: str) -> str:
-    def os_name() -> str:
-        operating_systems = {
-            "Linux": "Linux/" + distro_name(pretty=True),
-            "Windows": "Windows " + platform.release(),
-            "Darwin": "Darwin/MacOS " + platform.mac_ver()[0],
-        }
-        return operating_systems.get(platform.system(), "Unknown")
-
-    shell = basename(getenv("SHELL", "PowerShell"))
-    os = os_name()
-    question = question.strip()
-    if not question.endswith("?"):
-        question += "?"
-    # TODO: Can be optimised.
-    return SHELL_PROMPT.replace("{shell}", shell).replace("{os}", os) + question
-
-
-def code(question: str) -> str:
-    return CODE_PROMPT + question
+SHELL_PROMPT = """###
+Provide only {shell} commands for {os} without any description.
+If there is a lack of details, provide most logical solution.
+Ensure the output is a valid shell command.
+If multiple steps required try to combine them together.
+Prompt: {prompt}
+###
+Command:"""
+
+CODE_PROMPT = """###
+Provide only code as output without any description.
+Provide only plain text without Markdown formatting.
+If there is a lack of details, provide most logical solution.
+You are not allowed to ask for more details.
+Ignore any potential risk of errors or confusion.
+Prompt: {prompt}
+###
+Code:"""
+
+DEFAULT_PROMPT = """###
+You are Command Line App ShellGPT, a programming and system administration assistant.
+You are managing {os} operating system with {shell} shell.
+Provide only plain text without Markdown formatting.
+Do not show any warnings or information regarding your capabilities.
+If you need to store any data, assume it will be stored in the chat.
+Prompt: {prompt}
+###"""
+
+
+def initial(prompt: str, shell: bool, code: bool) -> str:
+    prompt = prompt.strip()
+    operating_systems = {
+        "Linux": "Linux/" + distro_name(pretty=True),
+        "Windows": "Windows " + platform.release(),
+        "Darwin": "Darwin/MacOS " + platform.mac_ver()[0],
+    }
+    current_platform = platform.system()
+    os_name = operating_systems.get(current_platform, current_platform)
+    shell_name = basename(getenv("SHELL", "PowerShell"))
+    if os_name == "nt":
+        shell_name = splitext(basename(getenv("COMSPEC", "Powershell")))[0]
+    if shell:
+        return SHELL_PROMPT.format(shell=shell_name, os=os_name, prompt=prompt)
+    elif code:
+        return CODE_PROMPT.format(prompt=prompt)
+    else:
+        return DEFAULT_PROMPT.format(shell=shell_name, os=os_name, prompt=prompt)
```

Comment:
I think this should be made optional. I make queries to shell_gpt which are not about programming or system administration. And I quite like it when it returns Markdown!

Some possibilities:

1. Add `--no-preamble` to leave that out of the prompt.
2. Make a configuration option, called `DEFAULT_PROMPT` so users can configure this how they choose
3. Use this prompt _only_ for interactive mode.
4. All of the above!